### PR TITLE
PXP-2464 fix/homepage chart REST API

### PR DESCRIPTION
**Pull request by qingyashu**
This PR change homepage chart to hit REST API instead of using relayer, and also enable `public` option for index page. 
- To set the chart nodes, a data commons should add `components.index.homepageChartNodes` for required nodes and chart labels. 
- To enable public index page, by setting `components.index.public`. 
- For backward compatibility, we still keep relayer to avoid crash. Until we gradually change all other data common' `gitops.json`/`${APP}.json` towards using REST API, then relayer-related code could be moved. 

### New Features
- Customizable public index page, by setting `components.index.public: true`. 

### Breaking Changes
- Change relayer to REST API for homepage chart: use `components.index.homepageChartNodes` for required nodes and chart labels.

### CODE REVIEW

**Comment by qingyashu** [comment]
Branch deployed in https://qingya.planx-pla.net/, now its index page is open to public.

File: src/Login/Login.jsx [hunk 1]
```diff
@@ -68,7 +78,7 @@ class Login extends React.Component {
           <hr className='login-page__separator' />
           <div className='body-typo'>{this.props.data.text}</div>
           {
-            this.props.providers.map(
+            this.props.providers && this.props.providers.map(
```
Comment (abgeorge7) [inline src/Login/Login.jsx]: is this necessary to add if you added default props?

Comment (qingyashu) [reply src/Login/Login.jsx]: Oh I changed the code yesterday and forgot to commit... let me update that! thanks

File: src/Index/page.jsx [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
 import { components } from '../params';
-import getProjectsList from './relayer';
+import getProjectNodeCounts from './utils';
 import './page.less';
 
 class IndexPageComponent extends React.Component {
   constructor(props) {
     super(props);
-    getProjectsList();
+    getProjectNodeCounts((res) => {
```
Comment (abgeorge7) [inline src/Index/page.jsx]: I think that this should go in `componentDidMount`, not the constructor.
https://reactjs.org/docs/react-component.html#constructor

**Pull request merged**